src/couch/src/couch_flags_config.erl

```diff
@@ -132,10 +137,7 @@ parse_flags_term(FlagsBin) ->
        [] ->
            lists:usort(Flags);
        _ ->
-           {error, {
-               "Cannot parse list of tags: ~n~p",
-               Errors
-           }}
+           {error, Errors}
```

Comment:
I ran the unit tests and noticed this change would need a corresponding change in the tests suite

```
in call from eunit_proc:run_group/2 (eunit_proc.erl, line 561)
**error:{assertEqual,[{module,couch_flags_config_tests},
              {line,115},
              {expression,"couch_flags_config : parse_flags_term ( LongBinary )"},
              {expected,{error,{"Cannot parse list of tags: ~n~p",
                                [{too_long,<<"aaaaaaaaaaaaaaaa"...>>}]}}},
              {value,{error,[{too_long,<<"aaaaaaaaaaaaaaaaaaaa"...>>}]}}]}
  output:<<"">>
```

Reply:
This should fix it:

```diff
diff --git a/src/couch/test/eunit/couch_flags_config_tests.erl b/src/couch/test/eunit/couch_flags_config_tests.erl
index ed7df1123..63fabfdde 100644
--- a/src/couch/test/eunit/couch_flags_config_tests.erl
+++ b/src/couch/test/eunit/couch_flags_config_tests.erl
@@ -98,10 +98,8 @@ test_config() ->
 
 parse_flags_term_test_() ->
     LongBinary = binary:copy(<<"a">>, ?MAX_FLAG_NAME_LENGTH + 1),
-    ExpectedError = {error, {"Cannot parse list of tags: ~n~p",
-       [{too_long, LongBinary}]}},
-    ExpectedUnknownError = {error,{"Cannot parse list of tags: ~n~p",
-       [{invalid_flag,<<"dddddddd">>}]}},
+    ExpectedError = {error, [{too_long, LongBinary}]},
+    ExpectedUnknownError = {error, [{invalid_flag,<<"dddddddd">>}]},
        [
                {"empty binary", ?_assertEqual(
                    [], couch_flags_config:parse_flags_term(<<>>))},

```

Reply:
Ah, yes, this was one area where I ended up refactoring things a bit as I was struggling with the pattern of carrying around a log formatting message as the return value through that whole parsing suite. Thanks for the heads up.

Reply:
Applied in 4ee64e5de